### PR TITLE
mcb-update-actions - Update matplotlib to latest version

### PR DESCRIPTION
**Pull request by M-casado**
## Ticket reference
NA

## Overall changes
- Updated matplotlib to latest changes, to check that GH actions resolve correctly.

## Future TO-DOs and checks
- [x] Check that the object's versions (i.e. ``meta:version``) are updated before creating the PR (see [GitHub Action](../.github/workflows/update_version_manifest.yml)). 
- [x] Check that the version manifest is up to date right before merging (see [documentation](../docs/releases/README.md#updating-the-version-manifest)).

### CODE REVIEW

**Comment by M-casado** [comment]
Outcome of the PR:

- Fixed conflicting dependencies (matplotlib)
- **Improved greatly the speed**, and lowered consumption, of GH workflows by adding caching to most Python actions.
- **Improved the validation reports** by adding summaries of the execution.
- Spent a few hours figuring out where the errors were coming from, and in the end traced them back again to https://github.com/elixir-europe/biovalidator/issues/57: 
   - Local validation works, but when some schemas are given to Biovalidator as referenced (``-r``), and these schemas also use APIs (e.g., keyword ``graphRestriction``), Biovalidator fails to compile the schemas and gives a cryptic message (e.g., `` {\"error\":\"Failed to compile schema: {}\"}`` - see https://github.com/EbiEga/ega-metadata-schema/actions/runs/9649515065/job/26613160012?pr=91)

In conclusion, the actions work, they work faster, and the reports are more comprehensive. But the local validation will keep failing with those schemas that use APIs.
